core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -328,6 +400,34 @@ private static void headerToXContent(XContentBuilder builder, String key, List<S
     }
 
     /**
+     * This method adds the current object value to the list of the headers, converting the value to String
+     * and flattening the name for objects and arrays.
+     */
+    @SuppressWarnings("unchecked")
+    private static void headerFromXContent(Map<String, List<String>> headers, String current, Object value) {
+        if (value instanceof Map) {
+            Map<String, ?> map = (Map) value;
+            for (Map.Entry<String, ?> entry : map.entrySet()) {
+                headerFromXContent(headers, current + "." + entry.getKey(), entry.getValue());
+            }
+        } else if (value instanceof List) {
+            List<?> list = (List) value;
+            for (int i = 0; i < list.size(); i++) {
+                Object item = list.get(i);
+                if (item instanceof String || item instanceof Number || item instanceof Boolean) {
+                    headerFromXContent(headers, current, item);
+                } else {
+                    headerFromXContent(headers, current + "." + i, item);
+                }
+            }
+        } else {
+            List<String> values = headers.getOrDefault(current, new ArrayList<>());
+            values.add(String.valueOf(value));
+            headers.put(current, values);
```

Comment:
after all, what would the parsed output of `SearchPhaseExecutionException` look like? I'm asking because that subclass prints out an array and potentially additional objects, I wonder how users would retrieve that additional info from the parsed ElasticsearchException.